Dashboard/app/js/lib/views/surveys/question-view.js

```diff
@@ -800,6 +805,120 @@ FLOW.QuestionView = FLOW.View.extend({
     var form = FLOW.selectedControl.get('selectedSurvey');
     return FLOW.permControl.canEditForm(form);
   }.property('FLOW.selectedControl.selectedSurvey'),
+
+  caddisflyTestSamples: function () {
+    var tests = FLOW.router.caddisflyResourceController.get('content'), unique = {}, distinct = [];
+    FLOW.selectedControl.set('selectedCaddisflyResource', null);
+    this.set('selectedCaddisflyTestSample', null);
+    tests.forEach(function (obj) {
+      if (typeof(unique[obj.sample]) == "undefined") {
+        distinct.push(obj);
+      }
+      unique[obj.sample] = 0;
+    });
+    return this.sortedList(distinct, 'sample');
+  }.property('FLOW.router.caddisflyResourceController.content'),
+
+  caddisflyTestNames: function () {
+    var tests = FLOW.router.caddisflyResourceController.get('content'), distinct = [];
+    FLOW.selectedControl.set('selectedCaddisflyResource', null);
+    this.set('selectedCaddisflyTestName', null);
+    if (this.get('selectedCaddisflyTestSample')) {
+      var sample = this.get('selectedCaddisflyTestSample'), unique = {};
+      var names = tests.filter(function (item) {
+        return item['sample'] === sample['sample'];
+      });
+      names.forEach(function (obj) {
+        if (typeof(unique[obj.name]) == "undefined") {
```

Comment:
same here you can use the `distinct` array to check whether a test name is already in there

```suggestion
        if (names.indexOf(obj.name) < 0) {
```

Reply:
same as above